build.sbt

```diff
@@ -2666,9 +2666,13 @@ def customFrgaalJavaCompilerSettings(targetJdk: String) = {
     // Ensure that our tooling uses the right Java version for checking the code.
     Compile / javacOptions ++= Seq(
       "-source",
-      frgaalSourceLevel,
-      "--enable-preview"
-    )
+      frgaalSourceLevel
+    ) ++
+    (if (Integer.parseInt(targetJdk) <= 21) {
```

Comment:
- Replacing this in #13244 with a better version
- that will use the JDK compiler, but also update `.enso-source*` files